robottelo/ui/discoveredhosts.py

```diff
@@ -106,11 +122,59 @@ def reboot_host(self, hostname):
             wait_for_ajax=False
         )
 
-    def update_org(self, hostnames, new_org):
+    def auto_provision(self, hostname):
+        """Auto provision the discovered host from dropdown"""
+        host = self.search(hostname)
+        if not host:
+            raise UIError(
+                'Could not find the selected discovered host "{0}"'
+                .format(hostname)
+            )
+        # Manually provision the host by selecting 'auto_provision' option
+        self.click(locators['discoveredhosts.dropdown'] % hostname)
+        self.click(locators['discoveredhosts.auto_provision'] % hostname)
+
+    def update_org_loc(self, hostnames, new_org=None, new_loc=None):
         """Update the default org or location for bulk of discovered hosts"""
         for host in hostnames:
             self.click(locators['discoveredhosts.select_host'] % host)
         self.click(locators['discoveredhosts.select_action'])
-        self.click(locators['discoveredhosts.assign_org'])
-        self.select(locators['discoveredhosts.select_org'], new_org)
+        if new_org:
+            self.click(locators['discoveredhosts.assign_org'])
+            self.select(locators['discoveredhosts.select_org'], new_org)
+        if new_loc:
+            self.click(locators['discoveredhosts.assign_loc'])
+            self.select(locators['discoveredhosts.select_loc'], new_loc)
         self.click(locators['discoveredhosts.bulk_submit_button'])
+
+    def provision_discoveredhost(self, hostgroup, org, loc, hostname,
+                                 new_name=None, parameters_list=None,
+                                 puppet_classes=None,
+                                 interface_parameters=None,
+                                 host_parameters=None, quick_create=False,
+                                 facts_page=False):
+        """Creates a host."""
+        if facts_page:
+            host = self.search(hostname)
+            host.click()
+            self.click(locators['discoveredhosts.select_action_facts'])
+            self.click(locators['discoveredhosts.provision_from_facts'])
+        else:
+            self.click(locators['discoveredhosts.provision'] % hostname)
+        self.select(
+            locators['discoveredhosts.select_modal_hostgroup'],
+            hostgroup
+        )
+        self.select(locators['discoveredhosts.select_modal_org'], org)
```

Comment:
@oshtaier : I tried assign_value but tests are failing due to this change. So I reverted the change.